src/main/java/org/opensearch/neuralsearch/processor/combination/ArithmeticMeanScoreCombinationMethod.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.processor.combination;
+
+import java.util.Objects;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+/**
+ * Abstracts combination of scores based on arithmetic mean method
+ */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class ArithmeticMeanScoreCombinationMethod implements ScoreCombinationMethod {
+
+    private static ArithmeticMeanScoreCombinationMethod INSTANCE = new ArithmeticMeanScoreCombinationMethod();
+
+    public static ArithmeticMeanScoreCombinationMethod getInstance() {
+        if (Objects.isNull(INSTANCE)) {
+            INSTANCE = new ArithmeticMeanScoreCombinationMethod();
+        }
+        return INSTANCE;
+    }
+
+    /**
+     * Arithmetic mean method for combining scores.
+     * cscore = (score1 + score2 +...+ scoreN)/N
+     *
+     * Zero (0.0) scores are excluded from number of scores N
+     */
+    @Override
+    public float combine(final float[] scores) {
+        float combinedScore = 0.0f;
+        int count = 0;
+        for (float score : scores) {
+            if (score >= 0.0) {
+                combinedScore += score;
+                count++;
+            }
+        }
+        return combinedScore / count;
```

Comment:
ack, I'll add more tests in next PRs

Reply:
Umm. I thought that divide by zero will throw an exception but I was wrong. For floating point number, it will be Infinity, -Infinity, or NaN. If those values are okay, I am fine.

Reply:
In such case we need to return 0.0, as this is the case when, if there are scores, we're skipping some of them and not increasing count. Let me update the method.